### ISSUE TITLE
Feature request: autosize

### ISSUE THREAD

**Issue by benman1** [shakedzy/dython#30]
It's easy to adjust _figsize_ yourself, and you can pass it to _associations()_, but it would be nice if this method could estimate an appropriate _figsize_ based on the number of columns.

**Comment by shakedzy** [comment]
Cool idea!

**Comment by Swish78** [comment]
I've implemented a solution to dynamically adjust the heatmap size based on the number of features in the dataset. The figure size scales proportionally, ensuring all features are displayed clearly, regardless of dataset size.

Would you like me to contribute this change to the project? Below are the before and after visuals for your reference:

Before:
![image](https://github.com/user-attachments/assets/44251b81-a1a7-4d82-9963-e0a0f36660c5)

After:
<img width="537" alt="Screenshot 2024-10-01 at 08 17 47" src="https://github.com/user-attachments/assets/675cf5f1-2b4e-4dbd-9a11-25763165f5a4">


This resolves the issue of overcrowded plots for larger datasets.

**Comment by benman1** [comment]
Hey @Swish78! The after screenshot looks great. I think you should go for it and create a PR! @shakedzy what do you say?

**Comment by shakedzy** [comment]
hey @Swish78 , i like the simplicity of it 💪 the only caveat here is that it doesn't take into account the length of the numbers displayed in the heat-map. I guess I'm brainstorming here with you - in the (rare?) case where a user wants use `fmt=.4f` or more figures in the heatmap, numbers will overlap. Is it something worth taking into account?

**Comment by Swish78** [comment]
Hi @shakedzy,

Thanks for your feedback! My initial approach was to consider both the precision of the numbers and the number of features when adjusting the figure size. However, I noticed this led to increased memory consumption, which might not be ideal, especially for larger plots.

That said, I assumed most users typically stick to a .2f format, which is why I leaned toward a simpler feature-based resizing. However, if memory consumption isn't a major concern, we could definitely account for both the number of features and the precision in the figure size configuration.

Let me know your thoughts!
